flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/aggfunctions/AvgAggFunction.scala

```diff
@@ -321,12 +317,11 @@ class DecimalAvgAggFunction extends AggregateFunction[BigDecimal] {
     if (value != null) {
       val v = value.asInstanceOf[BigDecimal]
       val accum = accumulator.asInstanceOf[DecimalAvgAccumulator]
+      accum.f0 = accum.f0.subtract(v)
+      accum.f1 -= 1L
       if (accum.f1 == 0) {
```

Comment:
This was intent to accommodate the "AggFunctionTestBase" for BigDecimal comparison (BigDecimal equals not only cares value but also the scale).  I overlooked that we can use bigDecimal.compareTo to compare the values. Will remove this and change the test cases.